third_party/move/move-vm/runtime/src/frame_type_cache.rs

```diff
@@ -69,43 +69,33 @@ pub(crate) struct FrameTypeCache {
     pub(crate) instantiated_local_ty_counts: Option<Rc<[NumTypeNodes]>>,
 }
 
-impl FrameTypeCache {
-    // note(inline):
-    // needs to always be inlined, closure will be optimized out in this case. When it gets inlined,
-    // LLVM also inlines `BTreeMap::entry()` with it to optimize, so the final instruction count is pretty big -
-    // do not inline the dependent functions.
-    #[inline(always)]
-    fn get_or<K: Copy + Ord + Eq, V, F>(
-        map: &mut BTreeMap<K, V>,
-        idx: K,
-        ty_func: F,
-    ) -> PartialVMResult<&V>
-    where
-        F: FnOnce(K) -> PartialVMResult<V>,
-    {
-        match map.entry(idx) {
-            std::collections::btree_map::Entry::Occupied(entry) => Ok(entry.into_mut()),
+macro_rules! get_or_insert {
```

Comment:
Just curious, does the compiler have trouble inlining the closure? (I ran into a similar issue with bin_op)

Reply:
Curious about this as well

Reply:
In my experience most of the time it inlines the closure just fine, but I've encountered a couple of cases when it doesn't for some reason. 
Here I'm making sure it always happens, because not inlining it here severely hits the performance. 

This macro also saves a couple of ASM instructions, because of simplified error handling.